File: app/src/main/java/me/toptas/fancyshowcasesample/CustomQueueActivity.java
package me.toptas.fancyshowcasesample;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import butterknife.BindView;
import butterknife.ButterKnife;
import me.toptas.fancyshowcase.FancyShowCaseQueue;
import me.toptas.fancyshowcase.FancyShowCaseView;
import me.toptas.fancyshowcase.OnViewInflateListener;

public class CustomQueueActivity extends BaseActivity {

    @BindView(R.id.btn_queue_1)
    Button mButton1;
    @BindView(R.id.btn_queue_2)
    Button mButton2;
    @BindView(R.id.btn_queue_3)
    Button mButton3;

    private FancyShowCaseQueue fancyShowCaseQueue;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_queue);
        ButterKnife.bind(this);

        final FancyShowCaseView fancyShowCaseView1 = new FancyShowCaseView.Builder(this)
                .title("First Queue Item")
                .focusOn(mButton1)
                .customView(R.layout.layout_my_custom_view, new OnViewInflateListener() {
                    @Override
                    public void onViewInflated(@NonNull View view) {
                        view.findViewById(R.id.btn_action_1).setOnClickListener(mClickListener);
                    }
                })
                .closeOnTouch(false)
                .build();

        final FancyShowCaseView fancyShowCaseView2 = new FancyShowCaseView.Builder(this)
                .title("Second Queue Item")
                .focusOn(mButton2)
                .customView(R.layout.layout_my_custom_view, new OnViewInflateListener() {
                    @Override
                    public void onViewInflated(@NonNull View view) {
                        view.findViewById(R.id.btn_action_1).setOnClickListener(mClickListener);
                    }
                })
                .closeOnTouch(false)
                .build();

        final FancyShowCaseView fancyShowCaseView3 = new FancyShowCaseView.Builder(this)
                .title("Third Queue Item")
                .focusOn(mButton3)
                .customView(R.layout.layout_my_custom_view, new OnViewInflateListener() {
                    @Override
                    public void onViewInflated(@NonNull View view) {
                        view.findViewById(R.id.btn_action_1).setOnClickListener(mClickListener);
                    }
                })
                .closeOnTouch(false)
                .build();

        fancyShowCaseQueue = new FancyShowCaseQueue()
                .add(fancyShowCaseView1)
                .add(fancyShowCaseView2)
                .add(fancyShowCaseView3);

        fancyShowCaseQueue.show();
    }


    View.OnClickListener mClickListener = new View.OnClickListener() {
        @Override
        public void onClick(View view) {
            Log.d("", "onClick: ");
            fancyShowCaseQueue.getCurrent().hide();
        }
    };

}

File: app/src/main/java/me/toptas/fancyshowcasesample/CustomQueueActivity.kt
/*
 * Copyright (c) 2018. Faruk Toptaş
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package me.toptas.fancyshowcasesample

import android.os.Bundle
import android.util.Log
import android.view.View

import kotlinx.android.synthetic.main.activity_queue.*
import me.toptas.fancyshowcase.FancyShowCaseQueue
import me.toptas.fancyshowcase.FancyShowCaseView
import me.toptas.fancyshowcase.listener.DismissListener
import me.toptas.fancyshowcase.listener.OnViewInflateListener

class CustomQueueActivity : BaseActivity() {

    private lateinit var queue: FancyShowCaseQueue

    private var mClickListener: View.OnClickListener = View.OnClickListener {
        Log.d("", "onClick: ")
        queue.current?.hide()
    }

    private var dismissListener = object:DismissListener{
        override fun onDismiss(id: String?) {
            Log.v("asd","dismiss")
        }

        override fun onSkipped(id: String?) {
            Log.v("asd","skipped")
        }

    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_queue)

        val fancyShowCaseView1 = FancyShowCaseView.Builder(this)
                .title("First Queue Item")
                .focusOn(btn_queue_1)
                .customView(R.layout.layout_my_custom_view, object : OnViewInflateListener {
                    override fun onViewInflated(view: View) {
                        view.findViewById<View>(R.id.btn_action_1).setOnClickListener(mClickListener)
                    }
                })
                .closeOnTouch(false)
                .dismissListener(dismissListener)
                .build()

        val fancyShowCaseView2 = FancyShowCaseView.Builder(this)
                .title("Second Queue Item")
                .focusOn(btn_queue_2)
                .customView(R.layout.layout_my_custom_view, object : OnViewInflateListener {
                    override fun onViewInflated(view: View) {
                        view.findViewById<View>(R.id.btn_action_1).setOnClickListener(mClickListener)
                    }
                })
                .closeOnTouch(false)
                .dismissListener(dismissListener)
                .build()

        val fancyShowCaseView3 = FancyShowCaseView.Builder(this)
                .title("Third Queue Item")
                .focusOn(btn_queue_3!!)
                .customView(R.layout.layout_my_custom_view, object : OnViewInflateListener {
                    override fun onViewInflated(view: View) {
                        view.findViewById<View>(R.id.btn_action_1).setOnClickListener(mClickListener)
                    }
                })
                .closeOnTouch(false)
                .dismissListener(dismissListener)
                .build()

        queue = FancyShowCaseQueue()
                .add(fancyShowCaseView1)
                .add(fancyShowCaseView2)
                .add(fancyShowCaseView3)

        queue.show()
    }

}
